### ISSUE TITLE
Cockpit needs ARIA / HTML5 landmarks

### ISSUE THREAD

**Issue by garrett** [cockpit-project/cockpit#8588]
For navigability as well as screen reading, Cockpit needs to have ARIA and/or HTML5 landmarks.

We use some of these tags and roles already, but Cockpit is currently lacking the most-important landmark `role="main"` for the content area. It could use more roles across the shell and on every page as well.

More info:
- [W3: HTML5 + ARIA landmarks](https://www.w3.org/TR/wai-aria-practices/examples/landmarks/HTML5.html)
- [Deque article on landmarks](https://dequeuniversity.com/assets/html/jquery-summit/html5/slides/landmarks.html)

**Comment by garrett** [comment]
Partially covered in:
- https://github.com/cockpit-project/cockpit/pull/8612/commits/d1339e244c07a18cb033971920b2ebdf15ec683b added role="main" to all content areas 
- https://github.com/cockpit-project/cockpit/pull/8612/commits/8b7ad439b4b965f2fb72afc7b161f22d0e3c994f use nav tag for navigation

Both are part of PR #8612.

**Comment by martinpitt** [comment]
What is left to be done here on a global level?

**Comment by garrett** [comment]
I need to look at Cockpit again in a screen reader.

Perhaps we can nicely ask @jgiardino to take a look at things too at some point. (But this is a huge ask, as Cockpit needs a lot of work still.)

**Comment by marusak** [comment]
We surely need a lot of improvements in accessibility. This issue should ideally list exact problems we should fix. Or even better have per page list of issues. Keeping open for now

**Comment by jgiardino** [comment]
Maybe January is a good time to review stuff together? ;-) 

For this specific issue, this [PF HTML Page demo](https://www.patternfly.org/v4/documentation/core/demos/page) illustrates some of the landmark roles being used. 

Also, PF uses aria-label for `<nav>` elements so that when navigating in a screen reader by region, the user understands the purpose of each navigation region. The text strings being used in PF for these aria-label attributes are:
- "Global" for the primary navigation
- "Local" for tabs (specifically, a single set of tabs in the main region of the page)
- "Breadcrumb" for breadcrumb navigation
- "Pagination" for pagination navigation that displays for a list of items

**Comment by marusak** [comment]
> Maybe January is a good time to review stuff together? ;-)

Absolutely, that would be great :)

**Comment by marusak** [comment]
This is too broad to be actionable.
We are getting every day better at this as we migrate to PF4 which makes us to use aria labels if they are needed, we think much more about accessibility when working on pages, on reviews we point out problems...
We can surely do more, but we should just scope it down like "fixing a11y issues on page foo", "fixing specific a11y problem over the cockpit", "using different component as it improves a11y"...
